### PR TITLE
Merge consul and instance-recovery features

### PR DESCRIPTION
**Pull request by MylesJP**
Summary of changes
- Move consul-related files into /instance-recovery
- Update consul unit tests to be general instance-recovery tests
- Remove 'sunbeam enable consul' function test from build-snap.yaml

### CODE REVIEW

**Review by hemanthnakkina — CHANGES_REQUESTED** [review]
My understanding, consul.py should constitute helper class to provide necessary information to InstanceRecoveryFeature class  + Consul classes related plans/Steps

The ConsulFeature class need not be a derived class of OpenStackControlPlaneFeature as it is more of a helper class. Do you see a need for the class to be a derived class?
If functions in ConsulFeature requires certain information like manifest, config etc they can be passed by the calling function in InstanceRecoveryFeature.
 
There are certain functions in consul.py that are not called by InstanceRecoveryFeature class and so can be removed like enable_command, disable_command.

**Comment by hemanthnakkina** [comment]
> @hemanthnakkina I updated the ConsulFeature object to consist of static methods that are simply called by InstanceRecovery when needed. I am noticing that the "Feature consul not found" message is still appearing then running `sunbeam enable instance-recovery`. Any thoughts on why this might be? I have this marked as draft for now.

Maybe its because manifests/edge.yml is not updated. Remove consul feature and add necessary charms under `instance-recovery`.
`instance-recovery` feature should not exist in rest of the yml files in manifest as this feature is supported only in edge.

**Review by gboutry — CHANGES_REQUESTED** [review]
Please, run the linter locally and apply the changes

File: sunbeam-python/sunbeam/features/instance_recovery/feature.py [hunk 1]
```diff
@@ -72,6 +81,8 @@ def manifest_attributes_tfvar_map(self) -> dict:
             }
         }
 
+        return consul_tfvar_map | instance_recovery_tfvar_map
```
Comment (hemanthnakkina) [inline sunbeam-python/sunbeam/features/instance_recovery/feature.py:84]: Need a deep copy of the dicts here. Currently consul_tfvar_map is overrided which leads to consul-channel not set in terraform tfvars.

**Review by hemanthnakkina — APPROVED** [review]

**Review by gboutry — APPROVED** [review]

**Comment by gboutry** [comment]
LGTM, nice work!

**Pull request merged**